src/lib/components/ng-http-loader.component.ts

```diff
@@ -52,12 +52,15 @@ export class NgHttpLoaderComponent implements OnDestroy, OnInit {
                 switchMap(() => showSpinner$.pipe(debounce(() => timer(this.debounceDelay))))
             ),
             showSpinner$.pipe(
-                switchMap(() => hideSpinner$.pipe(debounce(() => this.getVisibilityTimer())))
+                switchMap(() => hideSpinner$.pipe(debounce(() => this.getVisibilityTimer$())))
             ),
-            this.spinnerVisibilityService.visibilityObservable$,
+            this.spinnerVisibilityService.visibility$,
         )
-            .pipe(distinctUntilChanged())
-            .subscribe(h => this.handleSpinnerVisibility(h));
+            .pipe(
+                distinctUntilChanged(),
+                tap(h => this.updateVisibilityDuration(h))
+            )
+            .subscribe(h => this._isSpinnerVisible$.next(h));
```

Comment:
after you added updateVisibilityDuration you don't need ".subscribe(h => this._isSpinnerVisible$.next(h));" at all.
After you delete it you'll get the same Observable<boolean> in your subscriptions variable.

Reply:
Tried that, but tests failed because the observable needs to be initialized before the component bootstraps. That's why we have a `behaviorSubject` here. I couldn't manage to get rid of the subscription :/

Reply:
And having the subscription allows us to `unsubscribe` in `OnDestroy`

Reply:
if you get rid of subscribe method you won't need unsubscribe at all. This will be not your business

Reply:
>Tried that, but tests failed because the observable needs to be initialized before the component bootstraps.
You may use startWith(false) for observable stream

Reply:
The problem with what you want me to implement, is that every time the stream completes, `isSpinnerVisible$` receives a **new** Observable instance, and it just does not work with `OnPush`.

Reply:
Please fork and PR, so I can run tests locally. Because, implementing this as is simply does not play well with OnPush change detection.

Reply:
Could you please take and try code in a link I mentioned above? It's pretty same but related to your actual version. 
https://gist.github.com/byblik/0be812b2be750d1088d2dfe5af1ec416
We've got all components with OnPush change detection

Reply:
Tests failed: 58 :]

Reply:
which test exactly fails and what line?

Reply:
You'd better fork and run the tests. I have an implementation which nearly work with your suggestion, let me gist it to you in one minute

Reply:
Here you go : https://gist.github.com/mpalourdio/77ef20dca06aa8139924f9c3e5ae0505

Reply:
Time for beers now, will come back to this later. Thanks for your valuable feedback !

Reply:
you're welcome :)